src/Microsoft.Azure.WebJobs.Host/Loggers/FunctionInstanceLogger.cs

```diff
@@ -24,10 +26,27 @@ public Task<string> LogFunctionStartedAsync(FunctionStartedMessage message, Canc
         {
             string traceMessage = string.Format(CultureInfo.InvariantCulture, "Executing '{0}' (Reason='{1}', Id={2})", message.Function.ShortName, message.FormatReason(), message.FunctionInstanceId);
             Log(LogLevel.Information, message.Function, message.FunctionInstanceId, traceMessage);
-
+            if (message.TriggerDetails != null && message.TriggerDetails.Count != 0)
+            {
+                LogTemplatizedTriggerDetails(message);
+            }
             return Task.FromResult<string>(null);
         }
 
+        private void LogTemplatizedTriggerDetails(FunctionStartedMessage message)
+        {
+            var templateKeys = message.TriggerDetails.Select(entry => $"{entry.Key}: {{{entry.Key}}}");
+            string messageTemplate = "Trigger Details: " + string.Join(", ", templateKeys);
+            string[] templateValues = message.TriggerDetails.Select(entry => $"{entry.Value}").ToArray();
+            Log(LogLevel.Information, message.Function, messageTemplate, templateValues);
+        }
+
+        private void Log(LogLevel level, FunctionDescriptor descriptor, string message, params object[] args)
+        {
+            ILogger logger = _loggerFactory?.CreateLogger(LogCategories.CreateFunctionCategory(descriptor.LogName));
+            logger?.Log(level, 0, message, args);
```

Comment:
Is there a reason why we're creating the logger here with every invocation?

Reply:
Given the way the code is currently structured, the logger factory takes the function's name as an argument and the logs make it clear which function the logs stem from. It would be a larger change to, perhaps, create a logger at the outset and then inject each function's name into it without using the logger factory. This is because the `ILogger`, from what I can tell, does not allow setting the function name (also known category name). @brettsam to comment on how we could do this differently.

Reply:
Makes sense. This should reuse the logger internally, so we should be OK.